test/unit/gateways/global_collect_test.rb

```diff
@@ -35,6 +35,38 @@ def test_successful_authorize_and_capture
     assert_success capture
   end
 
+  def test_successful_authorize_and_capture_airline_fields
+    options = @options.merge(
+      airline_data: {
+        code: 111,
+        name: 'Spreedly Airlines',
+        flight_date: '20190810',
+        passenger_name: 'Randi Smith',
+        flight_legs: [
+          { arrival_airport: 'BDL',
+            origin_airport: 'RDU',
+            date: '20190810',
+            carrier_code: 'SA',
+            number: 596,
+            airline_class: 'ZZ'},
+          { arrival_airport: 'RDU',
+            origin_airport: 'BDL',
+            date: '20190817',
+            carrier_code: 'SA',
+            number: 597,
+            airline_class: 'ZZ'}
+        ]
+      }
+    )
+    stub_comms do
+      @gateway.purchase(@accepted_amount, @credit_card, options)
+    end.check_request do |endpoint, data, headers|
+      assert_equal 111, JSON.parse(data)['order']['additionalInput']['airlineData']['code']
+      assert_equal '20190810', JSON.parse(data)['order']['additionalInput']['airlineData']['flightDate']
+      assert_equal 2, JSON.parse(data)['order']['additionalInput']['airlineData']['flightLegs'].length
```

Comment:
These are some really nice `assert` statements! I don't think I've seen this structure before, but I quite like it. 🎉

Reply:
THANKS I cribbed it from Adyen https://github.com/activemerchant/active_merchant/pull/3460/files#diff-f2e3fbe51e77715bca1b237e7cb5e5bbR550 :D 
It seemed like a good fit for "make sure byzantine JSON is behaving itself".